R/infection_helpers.R

```diff
@@ -0,0 +1,136 @@
+#' Generate multiple infections with parameter distributions
+#'
+#' @param name An infection name from among `daedalus::epidemic_names`.
+#' @param samples The number of samples to generate.
+#'
+#' @param param_distributions A named list of `<distribution>` class
+#' objects provided by \pkg{distributional}, with names corresponding to
+#' `daedalus::infection_parameter_names`. These are used to generate `samples`
+#' draws from each distribution for the corresponding infection parameters.
+#' Arguments which vary by age are supported, with the drawn and scaled value
+#' treated as the median of the profile vector.
+#' See **Examples**.
+#'
+#' @param param_ranges An optional named list of two-element vectors, giving the
+#' ranges to which samples drawn using `param_distributions` should be rescaled.
+#' This allows easy rescaling of values drawn from bounded distributions (e.g.
+#' the Beta distribution).
+#' If passed, the list must have names corresponding to `param_distribution`
+#' names. When empty, the generated values are used without scaling.
+#' See **Examples**.
+#'
+#' @return A list of `<daedalus_infection>` objects.
+#' @export
+#'
+#' @examples
+#' # make 10 infection objects varying in R0, with a skewed distribution
+#' # scaled between 1.0 and 2.0
+#' make_infection_samples(
+#'   "influenza_2009",
+#'   samples = 3,
+#'   list(
+#'     r0 = distributional::dist_beta(2, 5)
+#'   ),
+#'   list(
+#'     r0 = c(0.1, 0.2)
+#'   )
+#' )
+make_infection_samples <- function(name,
+                                   param_distributions,
+                                   param_ranges = NULL,
+                                   samples = 100) {
+  # check inputs
```

Comment:
The logic through this function is hidden by the amount of parameter validation that goes on - perhaps separate it out into validation/generation so that the interesting bit is easier to spot